react-native/react/engine/index.js

```diff
@@ -118,8 +118,10 @@ class Engine {
   _generalIncomingRpc (method, param, response) {
     (this.generalListeners[method] || []).forEach(listener => {
       // TODO(mm) does it make sense to ever pass the response?
-      // It'll be weird if there are multiple listeners
-      listener(param)
+      // It'll be weird if there are multiple listeners.
+      // We should get the response callback to work the first
+      // time through and fail the second time.
+      listener(param, response)
```

Comment:
I only do this on incoming rpcs, not the notifications flow so you likely have to implement it there also: see https://github.com/keybase/client/blob/cjb/DESKTOP-6-pinentry-rpc/react-native/react/engine/index.js#L143